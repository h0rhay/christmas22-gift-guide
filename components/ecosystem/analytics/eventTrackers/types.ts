import { ClickEventActions } from "./eventsConstants";
import { EventTrackerType } from "./eventTracker";

export type GoogleAnalyticsType = {
  ga_event_name: string;
  ga_eventcategory?: string;
  ga_eventaction?: ClickEventActions;
  ga_eventlabel?: string;
};

export type CommonEventsTypes = {
  page_name: string;
  page_type?: string;
}

export type ViewEventTrackerType = {
  // specific for view event
  pageload_event?: string;
  page_load_time?: string; // not used anymore?
  page_subtype?: string;
  page_breadcrumb?: string
  page_category_name?: string
} & CommonEventsTypes
  & GoogleAnalyticsType
  & EventTrackerType

export type LinkEventTrackerType =
  CommonEventsTypes
  & GoogleAnalyticsType
  & EventTrackerType
  & Omit<EventTrackerType, 'event_name' | 'ga_event_name' | 'ga_eventcategory' | 'ga_eventaction'>;

export type DynamicViewEventTrackerType = {
  page_name: string;
}

export type DynamicLinkEventTrackerType = {
  ga_eventaction?: ClickEventActions;
  page_name: string;
  ga_eventlabel?: string;
}
