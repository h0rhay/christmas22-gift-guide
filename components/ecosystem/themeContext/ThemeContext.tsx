import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
