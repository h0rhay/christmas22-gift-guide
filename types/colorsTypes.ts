export type ColorGroupType = {
  surface: string;
  interface: string;
};

export type PrimitiveGreyColorType = {
  main: string;
  mid: string;
  light: string;
};

export type PrimitiveColorsType = {
  red: string;
  bubblegum: string;
  olive: string;
  yolk: string;
  sfYellow: string;
  milk: string;
  grey: PrimitiveGreyColorType;
};
