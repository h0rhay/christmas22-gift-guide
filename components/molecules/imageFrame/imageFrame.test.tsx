import { render } from "@testing-library/react";
import ImageFrame from "./imageFrame";
import { ThemeProvider } from "styled-components";
import theme from "styles/theme";
import ThemeContext from "../../ecosystem/themeContext/ThemeContext";
import { FrameOptionsType } from "types/";

const renderComponent = ({
  colorGroupName = theme.colorGroupNames[0],
  frameType = "mirror",
}) =>
  render(
    <ThemeProvider theme={theme}>
      <ThemeContext colorGroup={colorGroupName}>
        <ImageFrame
          src="/img/selfridges_logo.svg"
          alt="imageframe__image--alt"
          type={frameType as FrameOptionsType}
        />
      </ThemeContext>
    </ThemeProvider>
  );

let container: HTMLElement;
let frameType: FrameOptionsType;

describe("Image Frame", () => {
  it("Mirror Mobile should match snapshot", () => {
    const { container } = renderComponent({ frameType: "mirror" });
    expect(container).toMatchSnapshot();
  });

  it("Cloud Mobile should match snapshot", () => {
    const { container } = renderComponent({ frameType: "cloud" });
    expect(container).toMatchSnapshot();
  });

  it("Painting Desktop Should match snapshot", () => {
    const { container } = renderComponent({ frameType: "painting" });
    expect(container).toMatchSnapshot();
  });

  describe("Mirror shape", () => {
    beforeEach(() => {
      frameType = "mirror";
      ({ container } = renderComponent({ frameType }));
    });

    it("Container should have 3 children mobile and desktop and the image", () => {
      const svgContainer = container.getElementsByClassName(
        "imageframe__container"
      )[0];
      expect(svgContainer?.children.length).toBe(3);
    });

    it("Frame in/out border color must be colorgroup.interface and corner should be colorgroup.surface", () => {
      const colorGroupName = theme.colorGroupNames[0];
      const colorGroup = theme.colorsGroup[colorGroupName as string];

      const cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      const inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      const outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");
    });

    it("Mirror mobile display block and desktop display none", () => {
      const svgFrameMobile =
        container.getElementsByClassName("svg__frame--mobile")[0];
      expect(svgFrameMobile).toHaveStyle("display: block");

      const svgFrameDesktop = container.getElementsByClassName(
        "svg__frame--desktop"
      )[0];
      expect(svgFrameDesktop).toHaveStyle("display: none");
    });

    it("Frame in/out border color should change with theme ", () => {
      let colorGroupName = theme.colorGroupNames[0];
      let colorGroup = theme.colorsGroup[colorGroupName as string];

      let cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      let inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      let outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");

      colorGroupName = theme.colorGroupNames[1];
      colorGroup = theme.colorsGroup[colorGroupName as string];
      ({ container } = renderComponent({ frameType, colorGroupName }));
      cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");
    });
  });

  describe("Cloud shape", () => {
    beforeEach(() => {
      frameType = "cloud";
      ({ container } = renderComponent({ frameType }));
    });

    it("Container should have 3 children mobile and desktop and the image", () => {
      const svgContainer = container.getElementsByClassName(
        "imageframe__container"
      )[0];
      expect(svgContainer?.children.length).toBe(3);
    });

    it("Frame in/out border color must be colorgroup.interface and corner should be colorgroup.surface", () => {
      const colorGroupName = theme.colorGroupNames[0];
      const colorGroup = theme.colorsGroup[colorGroupName as string];

      const cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      const inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      const outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");
    });

    it("Mirror mobile display block and desktop display none", () => {
      const svgFrameMobile =
        container.getElementsByClassName("svg__frame--mobile")[0];
      expect(svgFrameMobile).toHaveStyle("display: block");

      const svgFrameDesktop = container.getElementsByClassName(
        "svg__frame--desktop"
      )[0];
      expect(svgFrameDesktop).toHaveStyle("display: none");
    });

    it("Frame in/out border color should change with theme ", () => {
      let colorGroupName = theme.colorGroupNames[0];
      let colorGroup = theme.colorsGroup[colorGroupName as string];

      let cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      let inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      let outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");

      colorGroupName = theme.colorGroupNames[1];
      colorGroup = theme.colorsGroup[colorGroupName as string];
      ({ container } = renderComponent({ frameType, colorGroupName }));
      cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");
    });
  });

  describe("Painting shape", () => {
    beforeEach(() => {
      frameType = "painting";
      ({ container } = renderComponent({ frameType }));
    });

    it("Container should have 3 children mobile and desktop and the image", () => {
      const svgContainer = container.getElementsByClassName(
        "imageframe__container"
      )[0];
      expect(svgContainer?.children.length).toBe(3);
    });

    it("Frame in/out border color must be colorgroup.interface and corner should be colorgroup.surface", () => {
      const colorGroupName = theme.colorGroupNames[0];
      const colorGroup = theme.colorsGroup[colorGroupName as string];

      const cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      const inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      const outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");
    });

    it("Mirror mobile display block and desktop display none", () => {
      const svgFrameMobile =
        container.getElementsByClassName("svg__frame--mobile")[0];
      expect(svgFrameMobile).toHaveStyle("display: block");

      const svgFrameDesktop = container.getElementsByClassName(
        "svg__frame--desktop"
      )[0];
      expect(svgFrameDesktop).toHaveStyle("display: none");
    });

    it("Frame in/out border color should change with theme ", () => {
      let colorGroupName = theme.colorGroupNames[0];
      let colorGroup = theme.colorsGroup[colorGroupName as string];

      let cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      let inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      let outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");

      colorGroupName = theme.colorGroupNames[1];
      colorGroup = theme.colorsGroup[colorGroupName as string];
      ({ container } = renderComponent({ frameType, colorGroupName }));
      cornerEl = container.getElementsByClassName("frame__corner")[0];
      expect(cornerEl).toHaveStyle("stroke: none");
      expect(cornerEl).toHaveStyle(`fill: ${colorGroup?.surface}`);

      inFrameEl = container.getElementsByClassName("frame__in")[0];
      expect(inFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(inFrameEl).toHaveStyle("fill: none");

      outFrameEl = container.getElementsByClassName("frame__out")[0];
      expect(outFrameEl).toHaveStyle(`stroke: ${colorGroup?.interface}`);
      expect(outFrameEl).toHaveStyle("fill: none");
    });
  });
});
