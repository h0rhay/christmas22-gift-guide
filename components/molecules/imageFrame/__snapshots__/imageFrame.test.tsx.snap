// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Frame Cloud Mobile should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="imageFrame__Container-sc-180q2f3-0 cA-DGOH imageframe__container undefined"
    >
      <div
        class="imageframe__image-container"
      >
        <div
          class="imageComponent__StyledImage-sc-y6629v-0 iprdOX image-component__container"
          type=""
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="imageframe__image--alt"
              class="imageframe__image"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(/img/selfridges_logo.svg); background-position: 0% 0%;"
            />
            <noscript />
          </div>
        </div>
      </div>
      <div
        class="frame__Container-sc-he37x9-0 gFwSUZ svg-frame-container svg__frame--mobile"
      >
        <svg
          class="svg__frame--mobile"
          viewBox="0 0 1180 1450"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="frame__corner"
            d="M0 0v1450h1180V0H0zm1042.1 924.8c25.9 224.3-107.3 301.3-243.1 315.5-51.3 83.4-114.7 137.6-208.3 137.6s-157.1-54.2-208.3-137.6c-135.7-14.2-265.4-82.9-243.1-315.5-48.6-48.1-72.4-117.4-72.4-200.7 0-81.7 25.4-154.7 72.4-200.7C103.2 300.7 246.7 222 382.4 207.8c51.3-83.4 114.7-137.6 208.3-137.6s157 54.3 208.3 137.7c135.7 14.2 275.8 93.7 243.1 315.5 55.2 52.7 71 121.1 71 200.7 0 83.1-15.7 148.6-71 200.7z"
          />
          <path
            class="frame__in"
            d="M50 725c0 76.5 24.5 157.6 74.7 207.2-23.1 240.1 110.8 311.1 250.9 325.7C428.5 1344 494 1400 590.6 1400s162.1-56 215-142.1c140.1-14.6 277.7-94.2 251-325.7 57.1-53.7 73.3-124.2 73.3-207.2 0-85.9-16.3-152.7-73.3-207.2 33.8-229-110.8-311.1-250.9-325.7C752.8 106 687.3 50 590.7 50s-162.1 56-215 142.1c-140.1 14.6-288.3 95.9-251 325.7C76.2 565.3 50 653.7 50 725z"
          />
          <path
            class="frame__out"
            d="M66.9 725c0 83.3 23.8 152.6 72.4 200.7-22.3 232.6 107.4 301.3 243.1 315.5 51.2 83.4 114.7 137.6 208.3 137.6s157-54.2 208.3-137.6c135.8-14.2 269-91.2 243.1-315.5 55.3-52.1 71-117.6 71-200.7 0-79.6-15.8-148-71-200.7C1074.8 302.5 934.7 223 799 208.8 747.7 125.4 684.3 71.1 590.7 71.1s-157 54.2-208.3 137.6c-135.7 14.2-279.2 92.9-243.1 315.6-47 46-72.4 119-72.4 200.7z"
          />
        </svg>
      </div>
      <div
        class="frame__Container-sc-he37x9-0 dXCvsx svg-frame-container svg__frame--desktop"
      >
        <svg
          class="svg__frame--desktop"
          viewBox="0 0 2020 1580"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="frame__corner"
            d="M0,0v1580h2020V0H0z M1805.3,996.1c-15.4,135.8-153.4,232.9-332.8,249.7c-43.9,4.1-86.5,1-126.4-6.5
    c-98.2,51.2-213,73.4-335.7,73.4s-237.5-22.2-335.6-73.4c-40,7.5-82.6,10.6-126.4,6.5C369,1229,230.9,1131.8,215.6,996
    C125.2,944.7,65.4,882.2,65.4,788.6s59.8-156.1,150.2-207.4C231,445.4,369,348.2,548.4,331.4c43.8-4.1,86.4-0.9,126.3,6.5
    c98.2-51.2,212.8-72.1,335.6-72.1s237.5,20.8,335.7,72.1c40-7.5,82.6-10.6,126.4-6.5c179.4,16.8,317.5,114,332.8,249.8
    c90.4,51.3,150.2,113.8,150.2,207.4C1955.4,882.3,1895.6,944.8,1805.3,996.1z"
          />
          <path
            class="frame__in"
            d="M1010,1330c124.7,0,241.3-21.9,341-74.7c40.6,7.7,83.8,9.9,128.4,5.7c182.3-17.3,323.5-116.5,339.1-256.6
    c91.8-52.9,151.6-118.4,151.6-215s-59.7-162.1-151.6-215c-15.6-140.2-156.9-239.4-339.1-256.8c-44.5-4.2-87.8-2-128.4,5.7
    c-99.7-52.8-216.3-73.3-341-73.3s-241.3,20.5-341,73.3c-40.6-7.7-83.8-9.9-128.4-5.7c-182.2,17.4-323.4,116.5-339,256.7
    C109.7,627.2,50,692.7,50,789.3s59.7,162.1,151.6,215c15.6,140.1,156.8,239.3,339.1,256.7c44.5,4.2,87.8,2,128.4-5.7
    C768.7,1308.1,885.3,1330,1010,1330z"
          />
          <path
            class="frame__out"
            d="M1010,1313.4c122.7,0,237.5-22.2,335.7-73.4c39.9,7.5,82.5,10.6,126.4,6.5c179.4-16.8,317.4-113.9,332.8-249.7
    C1895.2,945.5,1955,883,1955,789.3c0-93.6-59.8-156.1-150.2-207.4c-15.3-135.8-153.4-233-332.8-249.8c-43.8-4.1-86.4-1-126.4,6.5
    c-98.2-51.3-212.9-72.1-335.7-72.1s-237.4,20.9-335.6,72.1c-39.9-7.4-82.5-10.6-126.3-6.5c-179.4,16.8-317.4,114-332.8,249.8
    C124.8,633.2,65,695.7,65,789.3s59.8,156.1,150.2,207.4c15.3,135.8,153.4,233,332.8,249.8c43.8,4.1,86.4,1,126.4-6.5
    C772.5,1291.2,887.3,1313.4,1010,1313.4z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Image Frame Mirror Mobile should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="imageFrame__Container-sc-180q2f3-0 cA-DGOH imageframe__container undefined"
    >
      <div
        class="imageframe__image-container"
      >
        <div
          class="imageComponent__StyledImage-sc-y6629v-0 iprdOX image-component__container"
          type=""
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="imageframe__image--alt"
              class="imageframe__image"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(/img/selfridges_logo.svg); background-position: 0% 0%;"
            />
            <noscript />
          </div>
        </div>
      </div>
      <div
        class="frame__Container-sc-he37x9-0 gFwSUZ svg-frame-container svg__frame--mobile"
      >
        <svg
          class="svg__frame--mobile"
          viewBox="0 0 1180 1450"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="frame__corner"
            d="M0 0v1450h1180V0H0zm1113.7 1222.6c0 9.8-7.3 18.1-17.1 19.6-62.8 9.5-112.5 59.9-121.9 123.4-1.4 9.9-9.8 17.2-19.7 17.2H225c-9.9 0-18.2-7.4-19.7-17.2-9.3-63.5-59.1-113.8-121.9-123.4-9.8-1.5-17.1-9.7-17.1-19.6V225.4c0-9.8 7.3-18.1 17.1-19.6 62.8-9.5 112.5-59.9 121.9-123.4 1.4-9.9 9.8-17.2 19.7-17.2h730c10 0 18.3 7.3 19.7 17.2 9.3 63.5 59.1 113.8 121.9 123.4 9.8 1.5 17.1 9.7 17.1 19.6v997.2z"
          />
          <path
            class="frame__in"
            d="M966.3 50H213.7c-10.2 0-18.8 7.6-20.3 17.7-9.7 65.4-61 117.4-125.7 127.3-10.2 1.5-17.7 10-17.7 20.2v1019.6c0 10.2 7.5 18.7 17.7 20.2 64.7 9.9 116 61.8 125.7 127.3 1.5 10.1 10.1 17.7 20.3 17.7h752.7c10.2 0 18.8-7.6 20.3-17.7 9.7-65.5 61-117.4 125.7-127.3 10.1-1.5 17.7-10.1 17.7-20.2V215.2c0-10.2-7.5-18.7-17.7-20.2-64.7-9.9-116-61.8-125.7-127.3-1.5-10.1-10.1-17.7-20.4-17.7z"
          />
          <path
            class="frame__out"
            d="M955 66.2H225c-9.9 0-18.3 7.3-19.7 17.2-9.4 63.5-59.1 113.9-121.9 123.4-9.8 1.5-17.1 9.8-17.1 19.6v997.1c0 9.9 7.3 18.1 17.1 19.6 62.8 9.6 112.6 59.9 121.9 123.4 1.5 9.8 9.8 17.2 19.7 17.2h730c9.9 0 18.3-7.3 19.7-17.2 9.4-63.5 59.1-113.9 121.9-123.4 9.8-1.5 17.1-9.8 17.1-19.6V226.4c0-9.9-7.3-18.1-17.1-19.6-62.8-9.6-112.6-59.9-121.9-123.4-1.4-9.9-9.7-17.2-19.7-17.2z"
          />
        </svg>
      </div>
      <div
        class="frame__Container-sc-he37x9-0 dXCvsx svg-frame-container svg__frame--desktop"
      >
        <svg
          class="svg__frame--desktop"
          viewBox="0 0 2020 1580"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="frame__corner"
            d="M0,0v1580h2020V0H0z M1953.7,1155c0,9.9-7.4,18.2-17.2,19.7c-63.5,9.3-113.8,59.1-123.4,121.9
    c-1.5,9.8-9.7,17.1-19.6,17.1H226.4c-9.8,0-18.1-7.3-19.6-17.1c-9.5-62.8-59.9-112.5-123.4-121.9c-9.9-1.4-17.2-9.8-17.2-19.7V425
    c0-10,7.3-18.3,17.2-19.7c63.5-9.3,113.8-59.1,123.4-121.9c1.5-9.8,9.7-17.1,19.6-17.1h1567.1c9.8,0,18.1,7.3,19.6,17.1
    c9.5,62.8,59.9,112.5,123.4,121.9c9.9,1.4,17.2,9.8,17.2,19.7V1155z"
          />
          <path
            class="frame__in"
            d="M50,413.7v752.7c0,10.2,7.6,18.8,17.7,20.3c65.4,9.7,117.4,61,127.3,125.7c1.5,10.1,10.1,17.7,20.2,17.7
		h1589.6c10.2,0,18.7-7.5,20.2-17.7c9.9-64.7,61.8-116,127.3-125.7c10.1-1.5,17.7-10.1,17.7-20.3V413.7c0-10.2-7.6-18.8-17.7-20.3
		c-65.5-9.7-117.4-61-127.3-125.7c-1.5-10.2-10-17.7-20.2-17.7H215.2c-10.2,0-18.7,7.5-20.2,17.7c-9.9,64.7-61.8,116-127.3,125.7
		C57.6,394.8,50,403.4,50,413.7z"
          />
          <path
            class="frame__out"
            d="M66.2,425v730c0,9.9,7.3,18.3,17.2,19.7c63.5,9.4,113.9,59.1,123.4,121.9c1.5,9.8,9.8,17.1,19.6,17.1h1567.1
		c9.9,0,18.1-7.3,19.6-17.1c9.6-62.8,59.9-112.6,123.4-121.9c9.8-1.5,17.2-9.8,17.2-19.7V425c0-9.9-7.3-18.3-17.2-19.7
		c-63.5-9.4-113.9-59.1-123.4-121.9c-1.5-9.8-9.8-17.1-19.6-17.1H226.4c-9.9,0-18.1,7.3-19.6,17.1c-9.6,62.8-59.9,112.6-123.4,121.9
		C73.5,406.7,66.2,415,66.2,425z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Image Frame Painting Desktop Should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="imageFrame__Container-sc-180q2f3-0 cA-DGOH imageframe__container undefined"
    >
      <div
        class="imageframe__image-container"
      >
        <div
          class="imageComponent__StyledImage-sc-y6629v-0 iprdOX image-component__container"
          type=""
        >
          <div
            style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
          >
            <img
              alt="imageframe__image--alt"
              class="imageframe__image"
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(/img/selfridges_logo.svg); background-position: 0% 0%;"
            />
            <noscript />
          </div>
        </div>
      </div>
      <div
        class="frame__Container-sc-he37x9-0 gFwSUZ svg-frame-container svg__frame--mobile"
      >
        <svg
          class="svg__frame--mobile"
          viewBox="0 0 1180 1450"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="frame__corner"
            d="M0 0v1450h1180V0H0zm1111.6 725c0 358.8-55.8 431.6-15 611l-48.5 49.6c-200.2-35.6-339.8-1.9-457.6-1.7h-1.2c-117.8-.2-259-33.9-459.2 1.7l-44.3-44.8c40.9-179.4-17.6-257.2-17.6-616.2h.2c0-358.8 58.5-430.7 17.7-610.1L133.3 69c200.2 35.6 338.4.2 456.2 0h1.2c117.8.3 255.6 37.6 455.8 2l46 45.5c-40.9 179.4 19.3 249.5 19.3 608.5h-.2z"
          />
          <path
            class="frame__in"
            d="M589.5 50c-122 .3-258.3 36.5-465.5 0l-55.9 55.9c42.3 184-17.9 244.7-17.9 612.7H50c0 368.2 60.2 441.5 17.9 625.5l55.8 55.9c207.2-36.5 343.6-.3 465.5 0h1.2c121.9-.3 258.3-36.5 465.5 0l55.8-55.9c-42.2-184 17.9-257.1 17.9-625.1h.2c0-368.2-60.2-429.1-17.9-613.1L1056.2 50c-207.2 36.5-343.5.3-465.5 0h-1.2z"
          />
          <path
            class="frame__out"
            d="M589.5 69c-117.8.2-256 35.6-456.2 0l-47.2 45.5c40.8 179.4-17.7 251.3-17.7 610.1h-.2c0 359 58.5 436.8 17.6 616.2l44.3 44.8c200.2-35.6 341.4-1.9 459.2-1.7h1.2c117.8-.2 257.4-33.9 457.6 1.7l48.5-49.6c-40.8-179.4 15-252.2 15-611h.2c0-359-60.2-429.1-19.3-608.5l-46-45.5c-200.2 35.6-338-1.7-455.8-2h-1.2z"
          />
        </svg>
      </div>
      <div
        class="frame__Container-sc-he37x9-0 dXCvsx svg-frame-container svg__frame--desktop"
      >
        <svg
          class="svg__frame--desktop"
          viewBox="0 0 2020 1580"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="frame__corner"
            d="M0,0v1580h2020V0H0z M1954,789.2c0,0.3,0,1.1,0,1.3c-0.2,117.8-33.9,259,1.7,459.2l-44.8,44.3
    c-0.6-0.1-1.2-0.3-1.8-0.4l-0.6,0.5c-96.6-22-244.6-15.2-416.6-4.3c-37.7,2.4-76.5,5-116.2,7.5c-115.7,7.4-238.9,14.3-362.3,14.3
    v-0.2c-304.8,0-612.7-42.2-808.5-29.9c-33.7,2.2-64.1,6.1-90.5,12.1l-0.3-0.3c-0.7,0.2-1.4,0.3-2.1,0.5l-45.5-47.1
    c35.6-200.2,0.2-338.4,0-456.2v-1.2c0.3-117.8,37.6-255.6,2-455.8l45.5-46c0.6,0.1,1.2,0.3,1.8,0.4l0.5-0.6
    c91,20.7,228.7,15.5,389.1,5.3c56.8-3.6,116.5-7.9,177.9-11.8C790.6,273.9,903.4,268,1016,268v0.2c302.4,0,599.7,39.6,793.1,27.7
    c35.1-2.3,66.8-6.2,94.4-12.5l0.3,0.3c0.7-0.2,1.4-0.3,2-0.5l49.7,48.3C1920.1,531.8,1953.8,671.4,1954,789.2z"
          />
          <path
            class="frame__in"
            d="M50,790.3c0.3,121.9,36.5,258.3,0,465.5l55.9,55.8c184-42.3,535.9,17.9,903.9,17.9v0.2
		c368.2,0,720.3-60.2,904.3-17.9l55.9-55.8c-36.5-207.2-0.3-343.6,0-465.5l0,0c0-0.2,0-0.4,0-0.6c0-0.2,0-0.4,0-0.6l0,0
		c-0.3-121.9-36.5-258.3,0-465.5l-55.9-55.8c-184,42.2-535.9-17.9-903.9-18v-0.2c-368.2,0-720.3,60.2-904.3,17.9L50,323.6
		c36.5,207.2,0.3,343.5,0,465.5l0,0c0,0.2,0,0.4,0,0.6C50,789.9,50,790.1,50,790.3L50,790.3z"
          />
          <path
            class="frame__out"
            d="M69,790.4c0.2,117.8,35.6,256,0,456.2l45.5,47.1c179.4-40.8,542.5,17.7,901.3,17.7v0.2
		c359,0,715.6-58.5,895.1-17.6l44.8-44.3c-35.6-200.2-1.9-341.4-1.7-459.2l0,0c0-0.2,0-1,0-1.3l0,0c-0.2-117.8-33.9-257.4,1.7-457.6
		l-49.7-48.3c-179.4,40.8-531-15-889.8-15v-0.2c-359,0-720.3,60.2-899.7,19.3l-45.5,46c35.6,200.2-1.7,338-2,455.8V790.4z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
